### PR TITLE
Add error logging for deeply read

### PR DESCRIPTION
**Pull request by jfsoul**
## What does this change?
Adds some (temporary?) error logging for deeply read requests to ophan. This is to debug why we're failing to refresh this agent very often in production (without a redeploy).  I'll test it on CODE, in fact we may learn a lot from that and not need to roll this out necessarily.

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [ ] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [ ] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [ ] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [ ] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: onward/app/feed/DeeplyReadAgent.scala [hunk 1]
```diff
@@ -123,10 +121,13 @@ class DeeplyReadAgent(contentApiClient: ContentApiClient, ophanApi: OphanApi) ex
...
       }
-    fDeeplyReadItemsWithCapi.foreach { sequence =>
-      val mapDeeplyReadItems = sequence.filter(_.isDefined).map(_.get).toMap
-      deeplyReadItems = collection.mutable.Map(mapDeeplyReadItems.toSeq: _*)
-    }
+    fDeeplyReadItemsWithCapi.onComplete({
+      case Success(sequence) =>
+        val mapDeeplyReadItems = sequence.filter(_.isDefined).map(_.get).toMap
+        deeplyReadItems = collection.mutable.Map(mapDeeplyReadItems.toSeq: _*)
+      case Failure(exception) =>
```
Comment (nitro-marky) [inline onward/app/feed/DeeplyReadAgent.scala:128]: In case of failure will it retry or does it just fire an error?

**Review by nitro-marky — APPROVED** [review]
👍  for failure state

File: onward/app/feed/DeeplyReadAgent.scala (hunk 1 above)
Comment (jfsoul) [reply onward/app/feed/DeeplyReadAgent.scala:128]: Just logs the error message at the moment, if we discover what the issue is and decide that retries are the best way to solve it then we could look into that.

Comment (nitro-marky) [reply onward/app/feed/DeeplyReadAgent.scala:128]: 👍

**Comment by jfsoul** [comment]
Unneeded for the moment

**Pull request closed without merging**